packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts

```diff
@@ -1,147 +1,289 @@
-      }
-      try {
-        await sendLocationChangeEmails(
-          { ...evt, additionalInformation: metadata },
-          booking?.eventType?.metadata as EventTypeMetadata
-        );
-      } catch (error) {
-        console.log("Error sending LocationChangeEmails");
-      }
-    }
-  } catch {
-    throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
+export function getLocationForOrganizerDefaultConferencingAppInEvtFormat({
+  organizer,
+  loggedInUserTranslate: translate,
+}: {
+  organizer: {
+    name: string;
+    metadata: {
+      defaultConferencingApp?: NonNullable<UserMetadata>["defaultConferencingApp"];
+    } | null;
+  };
+  /**
+   * translate is used to translate if any error is thrown
+   */
+  loggedInUserTranslate: Awaited<ReturnType<typeof getTranslation>>;
+}) {
+  const organizerMetadata = organizer.metadata;
+  const defaultConferencingApp = organizerMetadata?.defaultConferencingApp;
+  if (!defaultConferencingApp) {
+    throw new UserError(
+      translate("organizer_default_conferencing_app_not_found", { organizer: organizer.name })
+    );
+  }
+  const defaultConferencingAppSlug = defaultConferencingApp.appSlug;
+  const app = getAppFromSlug(defaultConferencingAppSlug);
+  if (!app) {
+    throw new SystemError(`Default conferencing app ${defaultConferencingAppSlug} not found`);
+  }
+  const defaultConferencingAppLocationType = app.appData?.location?.type;
+  if (!defaultConferencingAppLocationType) {
+    throw new SystemError("Default conferencing app has no location type");
   }
+
+  const location = defaultConferencingAppLocationType;
+  const locationType = getEventLocationType(location);
+  if (!locationType) {
+    throw new SystemError(`Location type not found: ${location}`);
+  }
+
+  if (locationType.linkType === "dynamic") {
+    // Dynamic location type need to return the location as it is e.g. integrations:zoom_video
+    return location;
+  }
+
+  const appLink = defaultConferencingApp.appLink;
+  if (!appLink) {
+    throw new SystemError(`Default conferencing app ${defaultConferencingAppSlug} has no app link`);
+  }
+  return appLink;
+}
+
+export async function editLocationHandler({ ctx, input }: EditLocationOptions) {
+  const { newLocation, credentialId: conferenceCredentialId } = input;
+  const { booking, user: loggedInUser } = ctx;
+
+  const organizer = await UserRepository.findByIdOrThrow({ id: booking.userId || 0 });
+
+  const newLocationInEvtFormat = await getLocationInEvtFormatOrThrow({
+    location: newLocation,
+    organizer,
+    loggedInUserTranslate: await getTranslation(loggedInUser.locale ?? "en", "common"),
+  });
+
+  const evt = await buildCalEventFromBooking({
+    booking,
+    organizer,
+    location: newLocationInEvtFormat,
+    conferenceCredentialId,
+  });
+
+  const eventManager = new EventManager({
+    ...ctx.user,
+    credentials: await getAllCredentials({ user: ctx.user, conferenceCredentialId }),
```

Comment:
The result of `getAllCredentials` returns a duplicate of the conference credential. It works fine for now but I'm worried about any side effects in the future.

Reply:
I don't see any issues at the moment but should be handled though as it isn't expected to have duplicate credentials.  Can take it up in a followup.

Also, this is same as what was happening earlier just clubbed into `getAllCredentials`.